### PR TITLE
Return X-Vespa-Ignored-Fields if fields were ignored

### PR DESCRIPTION
**Pull request by bratseth**
I wasn't able to separate out the functional changes, sorry.

### CODE REVIEW

**Comment by bratseth** [comment]
The point of this is in DocumentV1ApiHandler.commit(..), 
and is mostly done in StructReader and VespaJsonDocumentReader.

File: document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java [hunk 1]
```diff
@@ -1006,17 +1009,65 @@ public void nonExistingFieldCausesException()  throws IOException{
...
+//                "    'nonexisting2': { " +
+//                "      'assign': {",
+//                "        'blocks': {",
+//                "          'a':[2.0,3.0],",
+//                "          'b':[4.0,5.0]",
+//                "        }",
+//                "      }",
+//                "    },",
+                "    'field2': { 'assign': 'value2' }",
+//                "    'nonexisting3': { 'assign': 'ignored value' }",
```
Comment (jonmv) [inline document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java:1058]: Intentionally commented out?

**Review by jonmv — APPROVED** [review]
I think maybe that one unit test wasn't done? Otherwise LGTM!

File: document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java (hunk 1 above)
Comment (bratseth) [reply document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java]: No, thanks!

File: document/src/main/java/com/yahoo/document/json/readers/VespaJsonDocumentReader.java [hunk 2]
```diff
@@ -185,6 +191,7 @@ private void addFieldPathUpdates(DocumentUpdate update, TokenBuffer buffer, Stri
             update.addFieldPathUpdate(fieldPathUpdate);
             buffer.next();
         }
+        return true; // TODO: Track fullyApplied for fieldPath updates
```
[the remark this answers is not in the document]
Comment (bratseth) [reply document/src/main/java/com/yahoo/document/json/readers/VespaJsonDocumentReader.java:194]: Yes.

File: container-core/src/main/java/com/yahoo/container/jdisc/state/StateHandler.java [hunk 3]
```diff
@@ -274,15 +274,11 @@ private ObjectNode buildJsonForSnapshot(String consumer, MetricSnapshot metricSn
...
-            case HEALTH_PATH:
-                return collapseHealthMetrics(snapshot);
-            case "all": // deprecated name
-            case METRICS_PATH:
-                return flattenAllMetrics(snapshot);
-            default:
-                throw new IllegalArgumentException("Unknown consumer '" + consumer + "'.");
-        }
+        return switch (consumer) {
+            case HEALTH_PATH -> collapseHealthMetrics(snapshot); // deprecated name
```
Comment (hmusum) [inline container-core/src/main/java/com/yahoo/container/jdisc/state/StateHandler.java:278]: I think this one got misplaces, should be on "all". And maybe deprecate and add note about removing in Vespa 9?

Comment (bratseth) [reply container-core/src/main/java/com/yahoo/container/jdisc/state/StateHandler.java:278]: Yes, added a better comment in https://github.com/vespa-engine/vespa/pull/24354

**Pull request merged**